--- a/drivers/i2c/busses/i2c-designware-baytrail.c	2016-10-03 01:24:33.000000000 +0200
+++ b/drivers/i2c/busses/i2c-designware-baytrail.c	2016-11-08 20:29:27.122528153 +0100
@@ -21,8 +21,9 @@
 
 #include "i2c-designware-core.h"
 
+// Semaphore registry is 0x7 for baytrail, 0x10E for cherrytrail
 #define SEMAPHORE_TIMEOUT	100
-#define PUNIT_SEMAPHORE		0x7
+#define PUNIT_SEMAPHORE		0x10E
 #define PUNIT_SEMAPHORE_BIT	BIT(0)
 #define PUNIT_SEMAPHORE_ACQUIRE	BIT(1)
 
